### PR TITLE
fix read() method

### PR DESCRIPTION
**Pull request by dsvoronin**
perform executes on app's main thread, so i just blocked it with waitFor(dumb)
now it's fixed

### CODE REVIEW

File: ui-testing-core/src/main/java/com/avito/android/test/espresso/action/TextViewReadAction.kt [hunk 1]
```diff
@@ -6,32 +6,40 @@ import android.support.test.espresso.matcher.ViewMatchers
...
 import org.hamcrest.Matcher
+import org.junit.Assert.assertFalse
 
-class TextViewReadAction(private val allowBlank: Boolean) : ViewAction {
+class TextViewReadAction : ViewAction {
 
-    lateinit var result: String
-        private set
+    companion object {
+        fun getResultVia(allowBlank: Boolean, perform: (TextViewReadAction) -> Unit): String {
```
Comment (nesterov-n) [inline ui-testing-core/src/main/java/com/avito/android/test/espresso/action/TextViewReadAction.kt]: Seems like method name is cut. I believe there should be `ViaSmth`

**Review by nesterov-n — DISMISSED** [review]

File: ui-testing-core/src/main/java/com/avito/android/test/espresso/action/TextViewReadAction.kt (hunk 1 above)
Comment (dsvoronin) [reply ui-testing-core/src/main/java/com/avito/android/test/espresso/action/TextViewReadAction.kt]: hmm, yep, definitely a naming problem
maybe getResult with comments on parameters will be better

File: ui-testing-core/src/main/java/com/avito/android/test/espresso/action/TextViewReadAction.kt [hunk 2]
```diff
@@ -6,32 +6,40 @@ import android.support.test.espresso.matcher.ViewMatchers
...
+            val action = TextViewReadAction()
+            return if (allowBlank) {
+                perform(action)
+                action.result ?: ""
+            } else {
+                waitFor(allowedExceptions = setOf(AssertionError::class.java)) {
```
Comment (dsvoronin) [inline ui-testing-core/src/main/java/com/avito/android/test/espresso/action/TextViewReadAction.kt]: user can't specify frequency and timeout for read(false), could be a problem
on other side let's not change signature before problem is validated

File: ui-testing-core/src/main/java/com/avito/android/test/espresso/action/TextViewReadAction.kt [hunk 3]
```diff
@@ -6,32 +6,45 @@ import android.support.test.espresso.matcher.ViewMatchers
...
+        /**
+         * @param perform use suitable driver to perform this view action
+         */
+        fun getResult(allowBlank: Boolean, perform: (TextViewReadAction) -> Unit): String {
+            val action = TextViewReadAction()
+            return if (allowBlank) {
+                perform(action)
+                action.result ?: ""
+            } else {
+                //FYI: will loop inside driver's loop here
```
Comment (dsvoronin) [inline ui-testing-core/src/main/java/com/avito/android/test/espresso/action/TextViewReadAction.kt:25]: found this "problem", not sure if we need to address it at all

**Review by nesterov-n — APPROVED** [review]

**Pull request merged**